### PR TITLE
Add Pure Storage FlashArray module to support NFS and S3 Offload

### PR DESCRIPTION
**Pull request by sdodsley**
##### SUMMARY
Add new module to support create and deletion of offload targets.
Current support is for one target of type NFS or S3.

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
purefa_offload

### CODE REVIEW

**Review by opslounge — APPROVED** [review]
shipit

**Comment by thedoubl3j** [comment]
will get eyes on this either later this evening or tomorrow AM.

File: lib/ansible/modules/storage/purestorage/purefa_offload.py [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+RETURN = r'''
+'''
+
+import re
+from distutils.version import LooseVersion
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.pure import get_system, purefa_argument_spec
+
+MIN_REQUIRED_API_VERSION = '1.16'
```
Comment (thedoubl3j) [inline lib/ansible/modules/storage/purestorage/purefa_offload.py:121]: Would this be worth documenting in the module's requirements field?

File: lib/ansible/modules/storage/purestorage/purefa_offload.py [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+---
+module: purefa_offload
+version_added: '2.8'
+short_description: Create, modify and delete NFS or S3 offload targets
+description:
+- Create, modify and delete NFS or S3 offload targets.
+- Only supported on Purity v5.2.0 or higher.
+- You must have a correctly configured offload network for offload to work.
+author:
+- Pure Storage Ansible Tesm (@sdodsley) <pure-ansible-team@purestorage.com>
```
Comment (thedoubl3j) [inline lib/ansible/modules/storage/purestorage/purefa_offload.py]: nit pick `Tesm` -> `Team`

File: lib/ansible/modules/storage/purestorage/purefa_offload.py (hunk 1 above)
Comment (sdodsley) [reply lib/ansible/modules/storage/purestorage/purefa_offload.py:121]: The module documentation declares the minimum version of the array OS required. This is translated through the REST API as the version listed. The user has no visibility of the API version used under the covers.

File: lib/ansible/modules/storage/purestorage/purefa_offload.py (hunk 2 above)
Comment (sdodsley) [reply lib/ansible/modules/storage/purestorage/purefa_offload.py]: Fixed

File: lib/ansible/modules/storage/purestorage/purefa_offload.py (hunk 1 above)
Comment (thedoubl3j) [reply lib/ansible/modules/storage/purestorage/purefa_offload.py:121]: roger that, thanks for the clarification @sdodsley

**Review by thedoubl3j — APPROVED** [review]

**Comment by gundalow** [comment]
@sdodsley @thedoubl3j @opslounge Thank you for this.
Merged into `devel` for release in Ansible 2.8

**Pull request merged**